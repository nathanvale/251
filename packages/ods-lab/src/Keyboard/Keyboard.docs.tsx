/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Keyboard, KeyboardProps } from "./Keyboard";

export const docs: ComponentDocs<KeyboardProps> = {
  category: "Experimental",
  componentName: "Keyboard",
  description: "",
  propDescriptions: {},
  migrationGuide: false,
  examples: {
    default: {
      Code: () => <Keyboard />,
    },
    additional: [],
  },
  snippets: [],
};
