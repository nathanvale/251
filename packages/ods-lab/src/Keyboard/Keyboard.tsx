/* eslint-disable react/no-array-index-key */
import React, { useEffect, useState, useRef } from "react";
import { KeyboardModel, KeyboardOptions, ActiveKeys } from "./KeyboardModel";
import { Chord } from "./Chord";

export interface KeyboardProps {
  leftHandKeys?: ActiveKeys | Chord;
  rightHandKeys?: ActiveKeys | Chord;
  options?: Partial<KeyboardOptions>;
}

function usePrevious(value: string) {
  const ref = useRef<string>();
  useEffect(() => {
    ref.current = value;
  });
  return ref.current;
}

//TODO error handling for out of range keys
export const Keyboard = ({
  rightHandKeys,
  leftHandKeys,
  options = {},
}: KeyboardProps) => {
  const [keyboardModel, setKeyboardModel] = useState<KeyboardModel>(
    new KeyboardModel(options)
  );

  const prevOptions = usePrevious(JSON.stringify(options));
  const prevLeftHandKeys = usePrevious(JSON.stringify(leftHandKeys));
  const prevRightHandKeys = usePrevious(JSON.stringify(rightHandKeys));

  useEffect(() => {
    const optionsChanged = prevOptions !== JSON.stringify(options);
    if (optionsChanged) {
      const km = new KeyboardModel(options);
      setKeyboardModel(km);
      km.playLeftHandKeys(leftHandKeys || {});
      km.playRightHandKeys(rightHandKeys || {});
    }
    if (prevLeftHandKeys !== JSON.stringify(leftHandKeys) && !optionsChanged) {
      keyboardModel.playLeftHandKeys(leftHandKeys || {});
    }
    if (
      prevRightHandKeys !== JSON.stringify(rightHandKeys) &&
      !optionsChanged
    ) {
      keyboardModel.playRightHandKeys(rightHandKeys || {});
    }
  }, [
    prevOptions,
    options,
    prevLeftHandKeys,
    leftHandKeys,
    prevRightHandKeys,
    rightHandKeys,
    keyboardModel,
  ]);

  const polygons = keyboardModel.getPolygons();
  const width = keyboardModel.getWidth();
  const height = keyboardModel.getHeight();
  const leftHandActiveKeys = keyboardModel.getLeftHandActiveKeys();
  const rightHandActiveKeys = keyboardModel.getRightHandActiveKeys();

  return (
    <>
      <svg width={width} height={height}>
        <g>
          {polygons &&
            polygons.map(({ polygon }, index) => {
              return [polygon && <polygon {...polygon} key={`p${index}`} />];
            })}
        </g>
        <g>
          {leftHandActiveKeys &&
            leftHandActiveKeys.map(({ polygon, text1, text2 }, index) => {
              return [
                polygon && <polygon {...polygon} key={`lhk${index}`} />,
                text1 && (
                  <text {...text1} key={`lhkt1${index}`}>
                    {text1.value}
                  </text>
                ),
                text2 && (
                  <text {...text2} key={`lhkt2${index}`}>
                    {text2.value}
                  </text>
                ),
              ];
            })}
          {rightHandActiveKeys &&
            rightHandActiveKeys.map(({ polygon, text1, text2 }, index) => {
              return [
                polygon && <polygon {...polygon} key={`rhk${index}`} />,
                text1 && (
                  <text {...text1} key={`rhkt1${index}`}>
                    {text1.value}
                  </text>
                ),
                text2 && (
                  <text {...text2} key={`rhkt2${index}`}>
                    {text2.value}
                  </text>
                ),
              ];
            })}
        </g>
      </svg>
    </>
  );
};

Keyboard.displayName = "Keyboard";

Keyboard.defaultProps = {};
