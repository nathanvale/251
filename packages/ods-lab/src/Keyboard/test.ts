/* eslint-disable no-console */
import { note, chord, /*Interval,*/ Chord as TeoriaChord } from "teoria";

// export function openChord(n) {
//   const note1 = note(n).interval("m7");
//   const note2 = note1.interval("M3");
//   const note3 = note2.interval("m2");
//   const note4 = note3.interval("M3");
//   return [note1, note2, note3, note4];
// }
interface Chord extends TeoriaChord {
  labels: () => Record<string, string>;
}
export function getChord(n: string) {
  const root = note(n); //?
  //const interval = Interval.toCoord("P5");

  const chord1 = chord(root)
    //.transpose(interval)
    .voicing(["P1", "M3", "P5", "M7", "M9"]);

  console.log(chord1.simple());
  console.log(chord1.toString());
  console.log(chord1.voicing());
  console.log(chord1.voicing().toString());
  console.log(chord1.bass().toString());
  console.log(chord1.toString());
  console.log(chord1.root);
  console.log(chord1.notes().toString());
  console.log(chord1.symbol);
  console.log(chord1.intervals);

  const customChord = chord1 as Chord;
  customChord.labels = () => {
    const voicing = chord1.voicing().toString().split(","); //?
    const notes = chord1.notes().toString().split(","); //?
    const labels = notes.reduce((previousValue, currentValue, currentIndex) => {
      return {
        ...previousValue,
        [currentValue.toUpperCase()]: voicing[currentIndex],
      };
    }, {});
    console.log(voicing); //?
    return labels; //?
  };

  customChord.simple = () => {
    const notes = chord1.notes().toString().split(",");
    return notes.map((note) => note.toUpperCase());
  };
  console.log(customChord.labels()); //?

  return customChord; //?
}

console.log(getChord("C3"));
