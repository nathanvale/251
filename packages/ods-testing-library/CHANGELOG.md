# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
