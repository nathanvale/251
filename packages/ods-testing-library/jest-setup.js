import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
