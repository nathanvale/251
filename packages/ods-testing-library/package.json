{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
