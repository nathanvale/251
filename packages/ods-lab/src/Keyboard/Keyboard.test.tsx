import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Keyboard } from "./Keyboard";

test("it renders a Keyboard", () => {
  const { container } = render(<Keyboard />);
  expect(container.firstChild).toMatchSnapshot();
});
