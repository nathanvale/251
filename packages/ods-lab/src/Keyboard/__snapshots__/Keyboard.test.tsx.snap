// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a Keyboard 1`] = `
<svg
  height="149"
  width="1229"
>
  <g>
    <polygon
      points="1,2 1,102 1,102 1,147 24.6,147 24.6,102 16.05,102 16.05,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="16.05,2 16.05,102 16.05,102 16.05,102 28.75,102 28.75,102 28.75,102 28.75,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="28.75,2 28.75,102 24.6,102 24.6,147 48.2,147 48.2,102 44.050000000000004,102 44.050000000000004,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="44.05,2 44.05,102 44.05,102 44.05,102 56.75,102 56.75,102 56.75,102 56.75,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="56.75,2 56.75,102 48.2,102 48.2,147 71.80000000000001,147 71.80000000000001,102 71.80000000000001,102 71.80000000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="71.80000000000001,2 71.80000000000001,102 71.80000000000001,102 71.80000000000001,147 95.4,147 95.4,102 85.75000000000001,102 85.75000000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="85.75,2 85.75,102 85.75,102 85.75,102 98.45,102 98.45,102 98.45,102 98.45,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="98.45,2 98.45,102 95.4,102 95.4,147 119,147 119,102 112.65,102 112.65,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="112.65,2 112.65,102 112.65,102 112.65,102 125.35000000000001,102 125.35000000000001,102 125.35000000000001,102 125.35000000000001,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="125.35,2 125.35,102 119,102 119,147 142.6,147 142.6,102 139.55,102 139.55,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="139.55,2 139.55,102 139.55,102 139.55,102 152.25,102 152.25,102 152.25,102 152.25,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="152.25000000000003,2 152.25000000000003,102 142.60000000000002,102 142.60000000000002,147 166.20000000000002,147 166.20000000000002,102 166.20000000000002,102 166.20000000000002,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="166.20000000000002,2 166.20000000000002,102 166.20000000000002,102 166.20000000000002,147 189.8,147 189.8,102 181.25000000000003,102 181.25000000000003,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="181.25,2 181.25,102 181.25,102 181.25,102 193.95,102 193.95,102 193.95,102 193.95,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="193.95000000000002,2 193.95000000000002,102 189.8,102 189.8,147 213.4,147 213.4,102 209.25,102 209.25,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="209.25,2 209.25,102 209.25,102 209.25,102 221.95,102 221.95,102 221.95,102 221.95,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="221.95000000000002,2 221.95000000000002,102 213.4,102 213.4,147 237,147 237,102 237,102 237,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="237,2 237,102 237,102 237,147 260.6,147 260.6,102 250.95,102 250.95,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="250.95,2 250.95,102 250.95,102 250.95,102 263.65,102 263.65,102 263.65,102 263.65,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="263.65000000000003,2 263.65000000000003,102 260.6,102 260.6,147 284.20000000000005,147 284.20000000000005,102 277.85,102 277.85,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="277.84999999999997,2 277.84999999999997,102 277.84999999999997,102 277.84999999999997,102 290.54999999999995,102 290.54999999999995,102 290.54999999999995,102 290.54999999999995,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="290.55000000000007,2 290.55000000000007,102 284.20000000000005,102 284.20000000000005,147 307.80000000000007,147 307.80000000000007,102 304.75000000000006,102 304.75000000000006,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="304.74999999999994,2 304.74999999999994,102 304.74999999999994,102 304.74999999999994,102 317.44999999999993,102 317.44999999999993,102 317.44999999999993,102 317.44999999999993,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="317.45,2 317.45,102 307.8,102 307.8,147 331.40000000000003,147 331.40000000000003,102 331.40000000000003,102 331.40000000000003,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="331.40000000000003,2 331.40000000000003,102 331.40000000000003,102 331.40000000000003,147 355.00000000000006,147 355.00000000000006,102 346.45000000000005,102 346.45000000000005,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="346.44999999999993,2 346.44999999999993,102 346.44999999999993,102 346.44999999999993,102 359.1499999999999,102 359.1499999999999,102 359.1499999999999,102 359.1499999999999,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="359.15,2 359.15,102 355,102 355,147 378.6,147 378.6,102 374.45,102 374.45,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="374.44999999999993,2 374.44999999999993,102 374.44999999999993,102 374.44999999999993,102 387.1499999999999,102 387.1499999999999,102 387.1499999999999,102 387.1499999999999,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="387.15000000000003,2 387.15000000000003,102 378.6,102 378.6,147 402.20000000000005,147 402.20000000000005,102 402.20000000000005,102 402.20000000000005,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="402.20000000000005,2 402.20000000000005,102 402.20000000000005,102 402.20000000000005,147 425.80000000000007,147 425.80000000000007,102 416.15000000000003,102 416.15000000000003,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="416.1499999999999,2 416.1499999999999,102 416.1499999999999,102 416.1499999999999,102 428.8499999999999,102 428.8499999999999,102 428.8499999999999,102 428.8499999999999,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="428.85,2 428.85,102 425.8,102 425.8,147 449.40000000000003,147 449.40000000000003,102 443.05,102 443.05,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="443.0499999999999,2 443.0499999999999,102 443.0499999999999,102 443.0499999999999,102 455.7499999999999,102 455.7499999999999,102 455.7499999999999,102 455.7499999999999,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="455.75000000000006,2 455.75000000000006,102 449.40000000000003,102 449.40000000000003,147 473.00000000000006,147 473.00000000000006,102 469.95000000000005,102 469.95000000000005,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="469.9499999999999,2 469.9499999999999,102 469.9499999999999,102 469.9499999999999,102 482.64999999999986,102 482.64999999999986,102 482.64999999999986,102 482.64999999999986,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="482.65,2 482.65,102 473,102 473,147 496.6,147 496.6,102 496.6,102 496.6,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="496.6,2 496.6,102 496.6,102 496.6,147 520.2,147 520.2,102 511.65000000000003,102 511.65000000000003,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="511.64999999999986,2 511.64999999999986,102 511.64999999999986,102 511.64999999999986,102 524.3499999999999,102 524.3499999999999,102 524.3499999999999,102 524.3499999999999,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="524.35,2 524.35,102 520.2,102 520.2,147 543.8000000000001,147 543.8000000000001,102 539.6500000000001,102 539.6500000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="539.6499999999999,2 539.6499999999999,102 539.6499999999999,102 539.6499999999999,102 552.3499999999999,102 552.3499999999999,102 552.3499999999999,102 552.3499999999999,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="552.35,2 552.35,102 543.8000000000001,102 543.8000000000001,147 567.4000000000001,147 567.4000000000001,102 567.4000000000001,102 567.4000000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="567.4000000000001,2 567.4000000000001,102 567.4000000000001,102 567.4000000000001,147 591.0000000000001,147 591.0000000000001,102 581.3500000000001,102 581.3500000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="581.3499999999999,2 581.3499999999999,102 581.3499999999999,102 581.3499999999999,102 594.05,102 594.05,102 594.05,102 594.05,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="594.05,2 594.05,102 591,102 591,147 614.6,147 614.6,102 608.25,102 608.25,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="608.25,2 608.25,102 608.25,102 608.25,102 620.95,102 620.95,102 620.95,102 620.95,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="620.95,2 620.95,102 614.6,102 614.6,147 638.2,147 638.2,102 635.15,102 635.15,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="635.1500000000001,2 635.1500000000001,102 635.1500000000001,102 635.1500000000001,102 647.8500000000001,102 647.8500000000001,102 647.8500000000001,102 647.8500000000001,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="647.85,2 647.85,102 638.2,102 638.2,147 661.8000000000001,147 661.8000000000001,102 661.8000000000001,102 661.8000000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="661.8000000000001,2 661.8000000000001,102 661.8000000000001,102 661.8000000000001,147 685.4000000000001,147 685.4000000000001,102 676.85,102 676.85,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="676.8500000000001,2 676.8500000000001,102 676.8500000000001,102 676.8500000000001,102 689.5500000000002,102 689.5500000000002,102 689.5500000000002,102 689.5500000000002,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="689.5500000000001,2 689.5500000000001,102 685.4000000000001,102 685.4000000000001,147 709.0000000000001,147 709.0000000000001,102 704.8500000000001,102 704.8500000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="704.8500000000001,2 704.8500000000001,102 704.8500000000001,102 704.8500000000001,102 717.5500000000002,102 717.5500000000002,102 717.5500000000002,102 717.5500000000002,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="717.55,2 717.55,102 709,102 709,147 732.6,147 732.6,102 732.6,102 732.6,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="732.6,2 732.6,102 732.6,102 732.6,147 756.2,147 756.2,102 746.5500000000001,102 746.5500000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="746.5500000000002,2 746.5500000000002,102 746.5500000000002,102 746.5500000000002,102 759.2500000000002,102 759.2500000000002,102 759.2500000000002,102 759.2500000000002,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="759.25,2 759.25,102 756.2,102 756.2,147 779.8000000000001,147 779.8000000000001,102 773.45,102 773.45,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="773.4500000000003,2 773.4500000000003,102 773.4500000000003,102 773.4500000000003,102 786.1500000000003,102 786.1500000000003,102 786.1500000000003,102 786.1500000000003,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="786.1500000000001,2 786.1500000000001,102 779.8000000000001,102 779.8000000000001,147 803.4000000000001,147 803.4000000000001,102 800.35,102 800.35,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="800.3500000000004,2 800.3500000000004,102 800.3500000000004,102 800.3500000000004,102 813.0500000000004,102 813.0500000000004,102 813.0500000000004,102 813.0500000000004,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="813.0500000000001,2 813.0500000000001,102 803.4000000000001,102 803.4000000000001,147 827.0000000000001,147 827.0000000000001,102 827.0000000000001,102 827.0000000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="827,2 827,102 827,102 827,147 850.6,147 850.6,102 842.05,102 842.05,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="842.0500000000004,2 842.0500000000004,102 842.0500000000004,102 842.0500000000004,102 854.7500000000005,102 854.7500000000005,102 854.7500000000005,102 854.7500000000005,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="854.75,2 854.75,102 850.6,102 850.6,147 874.2,147 874.2,102 870.0500000000001,102 870.0500000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="870.0500000000004,2 870.0500000000004,102 870.0500000000004,102 870.0500000000004,102 882.7500000000005,102 882.7500000000005,102 882.7500000000005,102 882.7500000000005,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="882.75,2 882.75,102 874.2,102 874.2,147 897.8000000000001,147 897.8000000000001,102 897.8000000000001,102 897.8000000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="897.8000000000001,2 897.8000000000001,102 897.8000000000001,102 897.8000000000001,147 921.4000000000001,147 921.4000000000001,102 911.7500000000001,102 911.7500000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="911.7500000000005,2 911.7500000000005,102 911.7500000000005,102 911.7500000000005,102 924.4500000000005,102 924.4500000000005,102 924.4500000000005,102 924.4500000000005,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="924.45,2 924.45,102 921.4000000000001,102 921.4000000000001,147 945.0000000000001,147 945.0000000000001,102 938.6500000000001,102 938.6500000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="938.6500000000005,2 938.6500000000005,102 938.6500000000005,102 938.6500000000005,102 951.3500000000006,102 951.3500000000006,102 951.3500000000006,102 951.3500000000006,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="951.35,2 951.35,102 945,102 945,147 968.6,147 968.6,102 965.55,102 965.55,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="965.5500000000006,2 965.5500000000006,102 965.5500000000006,102 965.5500000000006,102 978.2500000000007,102 978.2500000000007,102 978.2500000000007,102 978.2500000000007,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="978.25,2 978.25,102 968.6,102 968.6,147 992.2,147 992.2,102 992.2,102 992.2,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="992.2,2 992.2,102 992.2,102 992.2,147 1015.8000000000001,147 1015.8000000000001,102 1007.25,102 1007.25,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1007.2500000000007,2 1007.2500000000007,102 1007.2500000000007,102 1007.2500000000007,102 1019.9500000000007,102 1019.9500000000007,102 1019.9500000000007,102 1019.9500000000007,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1019.95,2 1019.95,102 1015.8000000000001,102 1015.8000000000001,147 1039.4,147 1039.4,102 1035.25,102 1035.25,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1035.2500000000007,2 1035.2500000000007,102 1035.2500000000007,102 1035.2500000000007,102 1047.9500000000007,102 1047.9500000000007,102 1047.9500000000007,102 1047.9500000000007,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1047.95,2 1047.95,102 1039.4,102 1039.4,147 1063,147 1063,102 1063,102 1063,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1063,2 1063,102 1063,102 1063,147 1086.6,147 1086.6,102 1076.95,102 1076.95,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1076.9500000000007,2 1076.9500000000007,102 1076.9500000000007,102 1076.9500000000007,102 1089.6500000000008,102 1089.6500000000008,102 1089.6500000000008,102 1089.6500000000008,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1089.65,2 1089.65,102 1086.6000000000001,102 1086.6000000000001,147 1110.2,147 1110.2,102 1103.8500000000001,102 1103.8500000000001,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1103.8500000000008,2 1103.8500000000008,102 1103.8500000000008,102 1103.8500000000008,102 1116.5500000000009,102 1116.5500000000009,102 1116.5500000000009,102 1116.5500000000009,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1116.55,2 1116.55,102 1110.2,102 1110.2,147 1133.8,147 1133.8,102 1130.75,102 1130.75,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1130.750000000001,2 1130.750000000001,102 1130.750000000001,102 1130.750000000001,102 1143.450000000001,102 1143.450000000001,102 1143.450000000001,102 1143.450000000001,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1143.4500000000003,2 1143.4500000000003,102 1133.8000000000002,102 1133.8000000000002,147 1157.4,147 1157.4,102 1157.4,102 1157.4,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1157.4,2 1157.4,102 1157.4,102 1157.4,147 1181,147 1181,102 1172.45,102 1172.45,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1172.450000000001,2 1172.450000000001,102 1172.450000000001,102 1172.450000000001,102 1185.150000000001,102 1185.150000000001,102 1185.150000000001,102 1185.150000000001,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1185.15,2 1185.15,102 1181,102 1181,147 1204.6,147 1204.6,102 1200.45,102 1200.45,2"
      style="fill: #F2F2EF; stroke: #39383D; stroke-width: 1;"
    />
    <polygon
      points="1200.450000000001,2 1200.450000000001,102 1200.450000000001,102 1200.450000000001,102 1213.150000000001,102 1213.150000000001,102 1213.150000000001,102 1213.150000000001,2"
      style="fill: #39383D; stroke: #39383D; stroke-width: 1;"
    />
  </g>
  <g />
</svg>
`;
